File: gpt-3.5/template_context_v2/data_construct_detach_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:data construct detach clause,V:2.7-3.3
int test1(){
    int err = 0;
    int a[N], b[N], c[N];
    int i;

    // Initialize arrays
    for(i=0; i<N; i++){
        a[i] = i;
        b[i] = i;
    }

    // Compute c[i] = a[i] + b[i]
    #pragma acc data detach(a, b)
    {
        #pragma acc kernels
        for(i=0; i<N; i++){
            c[i] = a[i] + b[i];
        }
    }

    // Verify result
    for(i=0; i<N; i++){
        if(c[i] != 2*i){
            err = 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/template-context-full/data_construct_detach_clause.f90
#ifndef T1
!T1:data construct detach clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(KIND=8),DIMENSION 1000:: a, b,c
        REAL(KIND=8),DIMENSION(true)  :: sm, sm2, sum, sum2
	REAL(KIND=8):: temp
      a = (/ (i, i = 1, 1000) /)
      b = (/ (i, i = 1, 1000) /)
      c = (/ (i, i = 1, 1000) /)

      ! $acc enter data create(sm(1:1000)))
      !$acc data create(sm(1:1000)) copyin(a(1:1000), b(1:1000), c(1:1000))
        !$acc parallel
          !$acc loop
          DO m = 1, 100
            !$acc loop vector
            DO n = 1, 10
              sm(m) = sm(m) + a((m*10)+n)
            END DO
            !$acc loop vector
            DO n = 1, 10
              c((m*10)+n) = b((m*10)+n) + sm(m)
            END DO
          END DO
        !$acc end parallel
      !$acc end detach(sm, acc_present(1:1000))

      !$acc parallel
        !$acc loop
        DO m = 1, 100
          !$acc loop vector
          DO n = 1, 10
            sm2(m) = sm2(m) + a((m*10)+n)
          END DO
          !$acc loop vector
          DO n = 1, 10
            sum(m) = sum(m) + c((m*10)+n)
          END DO
          !$acc loop vector
          DO n = 1, 10
            sum2(m) = sum2(m) + b((m*10)+n)
          END DO
          IF (sm2(m) .neqv. sum(m) .or. sm2(m) .neqv. sum2(m)) THEN
            errors = errors + 1
          END IF
        END DO
      !$acc end parallel

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM